src/main/java/io/vertx/core/net/HostAndPort.java

```diff
@@ -55,6 +60,24 @@ static HostAndPort authority(String host) {
     return authority(host, -1);
   }
 
+  /**
+   * Convert a {@link SocketAddress} to a {@link HostAndPort}.
+   * If the socket address is an {@link InetSocketAddress}, the hostString and port are used.
+   * Otherwise {@code null} is returned.
+   *
+   * @param socketAddress The socket address to convert
+   * @return The converted instance or {@code null} if not applicable.
+   */
+  @Nullable
+  @GenIgnore(GenIgnore.PERMITTED_TYPE)
```

Comment:
this should use instead directly inet socket address and not assume we accept any socket address and instead return null

Reply:
@vietj This method is added to simplify the caller (no need to add if-else to check if the `SocketAddress` is an `InetSocketAddress` in caller side)
If making the parameter to be `InetSocketAddress` then this method is less useful and I'd rather move this method to somewhere else.
(I'm still thinking passing `InetSocketAddress` directly to `SslChannelProvider` is the simplest and having minimal overhead)
What do you think?

Reply:
the issue is that we prefer to avoid methods that perform the null check, the caller should not rely on the method to return null when passing an inapropriate argument, instead it should check the argument and perform the work itself, eventually it can be in an internal util class, but we should not encourage nor promote this on a public API.

Reply:
@vietj You didn't object my previous comment https://github.com/eclipse-vertx/vert.x/pull/5346#discussion_r1794552108 about returning the nullable on public API so I assumed you agree on that:
> If use `HostAndPort`, I may need to add a conversion method like `@Nullable static HostAndPort socketHostAndPort(SocketAddress address);` to simplify the conversion, is it OK to put it in `HostAndPort` interface (which will become a new external API)?

I don't understand your expectation and reason behind it now.
You asked to use `HostAndPort` instead of `SocketAddress`, while not allowing adding a method that returns nullable for public API.
If like you said, the caller should check the argument, then it will become:
```
// in HttpServerWorker.configurePipeline
SocketAddress remoteAddress = ch.remoteAddress();
HostAndPort remoteHostAndPort = (remoteAddress instanceof InetSocketAddress ? HostAndPort.fromSocketAddress(remoteAddress) : null);
pipeline.addLast("ssl", sslChannelProvider.createServerHandler(remoteHostAndPort));

// in NetServerImpl.NetServerWorker.configurePipeline
SocketAddress remoteAddress = ch.remoteAddress();
HostAndPort remoteHostAndPort = (remoteAddress instanceof InetSocketAddress ? HostAndPort.fromSocketAddress(remoteAddress) : null);
ch.pipeline().addLast("ssl", sslChannelProvider.createServerHandler(remoteHostAndPort));

// in NetSocketImpl.upgradeToSsl
SocketAddress remoteAddress = chctx.channel().remoteAddress();
HostAndPort remoteHostAndPort = (remoteAddress instanceof InetSocketAddress ? HostAndPort.fromSocketAddress(remoteAddress) : null);
sslHandler = sslChannelProvider.createServerHandler(remoteHostAndPort);
```
This will result in duplicated code and increase maintenance effort, so I don't think it is a good option.

Another option may be moving this method to an internal API, e.g., move to `SslChannelProvider` or other internal util class.
If moving to `SslChannelProvider`, then why do we need brother changing `public ChannelHandler createServerHandler(java.net.SocketAddress remoteAddress)` to `public ChannelHandler createServerHandler(HostAndPort remoteAddress)`?
If move to another internal util class, could you advice a proper place to put this `SocketAddress` to `HostAndPort` conversion method that may return null?

Reply:
I looked around and find `HttpUtils`, I'll try move this method to there.